Finish section "Limits" in iset.mm

jkingdon opened the pull request:
A lot of this is pretty straightforward given the now-familiar differences between iset.mm and set.mm over `seq` and similar topics.

Includes climcvg1n which represents taking previous "a Cauchy sequence converges' results in iset.mm and updating to (a) use the absolute value notation, (b) use the `~~>` notation for convergence, (c) extend from real numbers to complex numbers.

Includes a number of results related to Cauchy sequences (even in set.mm there are a few different expressions and theorems to convert between them - in iset.mm there is the additional complication that "Cauchy sequence" sometimes means the climcvg1n-style concept which includes a rate of convergence, and sometimes just means the set.mm expressions).

jkingdon committed:
Add clim2iser2 to iset.mm

Based on clim2ser2 from set.mm with the usual adjustments for the
differences in seq between iset.mm and set.mm.

jkingdon committed:
Add iiserex to iset.mm

Based on iserex in set.mm with (minimal) changes for the differences
in seq.

jkingdon committed:
Add iseqhomo to iset.mm

Intuitionized version of seqhomo, differing in the usual ways for
differences in how seq works.

jkingdon committed:
Add iseqdistr to iset.mm

Based on seqdistr from set.mm with the usual changes for differences in
the iset.mm version of seq .

jkingdon committed:
Add iisermulc2 to iset.mm

This is taken from isermulc2 in set.mm. The only change to the statement
of the theorem is in the syntax of seq .

The proof needs to be changed modestly for differences in the iset.mm
iseqdistr compared with set.mm seqdistr but that it straightforward.

jkingdon committed:
Add climlec2 to iset.mm

Copied without change from set.mm.

jkingdon committed:
Add serige0 to iset.mm

Based on serge0 from set.mm with some of the usual changes for
differences in seq .

jkingdon committed:
Add serile to iset.mm

Based on serle from set.mm.

jkingdon committed:
Add iserile to iset.mm

Similar to iserle from set.mm with the usual adjustments for differences
in seq .

jkingdon committed:
Add iserige0 to iset.mm

Based on iserge0 from set.mm with only minor changes needed for
differences in seq syntax.

jkingdon committed:
Add climub to iset.mm

Copied without change from set.mm.

jkingdon committed:
Add climserile to iset.mm

Based on climserle from set.mm with the usual changes for differences in
seq .

jkingdon committed:
Add isershft to mmil.html

This seems like a certain amount of work and maybe we don't have to do
it just yet looking at the usages.

jkingdon committed:
Add isercoll to mmil.html

This is one which doesn't look super easy. Also includes isercoll2 , its
corollary.

jkingdon committed:
Add climsup to mmil.html

I'm usually reluctant to make the categorical statements about "this one
cannot be proved" but hopefully what I have said here isn't going out
too much on a limb.

jkingdon committed:
Add climcau to iset.mm

Copied without change from set.mm. I'll make a follow up change to add a
few words to the comment about "Cauchy sequence" as a concept.

jkingdon committed:
Expand comment of climcau in iset.mm

This theorem itself holds in iset.mm just like in set.mm. But it
seems worth mentioning the converse and other ways of defining "Cauchy
sequence", at least briefly.

jkingdon committed:
Add climbdd to mmil.html

This would be fun to prove, but not doing it yet.

jkingdon committed:
Add climrecvg1n to iset.mm

This is a version of cvg1n which is updated to use absolute value and
the ~~> notation.

jkingdon committed:
Add climcvg1n to iset.mm

This takes climrecvg1n and extends it to complex numbers (which turns
out not to be too hard, by applying it to the real and imaginary
parts of each term).

Includes lemma climcvg1nlem (which is the bulk of the proof, just
separate so we can give names to some sequences we define).

Update comment of climcau to point to climcvg1n .

jkingdon committed:
Add caucvgb to mmil.html

I'm not thinking of a biconditional which would make sense so just list
two theorems which are sort of but not really converses of each
other.

jkingdon committed:
Add cau3lem and cau3 to iset.mm

Copied without change from set.mm.

jkingdon committed:
Add cau4 to iset.mm

Copied without change from set.mm.

jkingdon committed:
Add caubnd2 to iset.mm

Copied without change from set.mm.

Update mmil.html discussion of caubnd .

jkingdon committed:
Add climcaucn to iset.mm

The proof is based on part of the set.mm proof of caucvgb .

Update caucvgb entry in mmil.html to point to this.

jkingdon committed:
Add serif0 to iset.mm

Based on serf0 in set.mm but with a change for the syntax of seq .

The proof contains the usual seq related changes and also changing
caucvgb to climcaucn (which works fine because the set.mm proof had been
using only this direction of the biconditional).

jkingdon committed:
Add iseralt to mmil.html

This one looks hard, both because it is a long proof, but also because
at least one key step doesn't work for iset.mm.

nmegill commented:
The iset-discouraged check fails.
-New usage of "4syl" is discouraged (11 uses).
+New usage of "4syl" is discouraged (12 uses).

jkingdon committed:
Regenerate iset-discouraged

Adds one use of 4syl (in serif0 ) which is fine (4syl is only
discouraged to avoid slowing down the minimizer).

jkingdon commented:
> The iset-discouraged check fails.
> -New usage of "4syl" is discouraged (11 uses).
> +New usage of "4syl" is discouraged (12 uses).

Fixed now.
